db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj

```diff
@@ -2353,9 +2353,25 @@ public class SQLParser
      * CREATE TABLE X AS [SELECT A, B, C FROM Y] WITH DATA
      */
    public String parseQueryString(String src) {
-       boolean inSection = false;
-       StringBuilder buf = new StringBuilder();
+        boolean inSection = false;
+        StringBuilder buf = new StringBuilder();
+
+        // Use a non-printable character as an end-of-hint marker.
+        char endOfHintMarker = 0x01;
+
+        // Strip out comments starting with --, except --splice-properties hints.
+        src = src.replaceFirst("(--)(?i)(?!splice-properties)(.*?)(\\n|\\r|\\r\\n)", " ");
```

Comment:
The fix may not work for some complex scenarios:
1) If "--" is inside the C-style comment, for example
create table v1 as 
select a1, b1 from t1 /* c-style comment starts with an embedded -- comment */
where a1 > 3 with data;
We could mistakenly remove the closing C-style comment. Maybe we should switch the order of processing for C-style comment and -- comment.
2) there could be more than one splice-properties hint, and replaceFirst is not sufficient.

Reply:
Thanks @yxia92!  I'm updating the fix.

Reply:
Switching the order of processing the C-style comment and dash dash comment does not work either. For example:
```
create table v1 as
select a1, b1 from t1 /* c-style comment 
, t2 -- dash dash comment with an embedded */
*/
where a1 > 3 with data;
```
However, I don't see an easy way to cover this case except for writing a parser using maybe javacc. Since it is a rare case, I'm OK with proceeding the fix and add this test case to DB-8575 also.